docs/latex/Inhalt/03_Inhalt/07_refactoring.tex

```diff
@@ -8,10 +8,10 @@ \subsection{Duplicated Code} \label{sec:duplicatedCode}
+Die Klasse \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/0-quickie-plugin-scraper/src/main/java/org/pinkcrazyunicorn/quickie/plugins/scraper/HensslerScraper.java}{\code{HensslerScraper}} hat eine stark erhöhte Zeilen- und Methodenanzahl im Vergleich zu anderen Klassen des Projektes. In dieser Klasse wird implementiert, wie Daten aus Webseite extrahiert werden und was extrahiert werden soll. Hier ist das Single Responsibility Principle verletzt, da die Klasse sowohl implementiert, wie ein Rezept aus einer Rezeptseite extrahiert wird, aber auch wie alle Rezeptseiten gefunden werden und zugleich dazu jeweils das \enquote{Wie} sowie das \enquote{Was}. Aufgrund dieser Verletzung des Single Responsibility Principles und der großen Menge an Code ist die Klasse sehr unübersichtlich. So wird zum Beispiel in \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/0-quickie-plugin-scraper/src/main/java/org/pinkcrazyunicorn/quickie/plugins/scraper/HensslerScraper.java#L22}{Zeile 22} eine Konstante definiert, welche erst über 100 Zeilen später in der Methode \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/0-quickie-plugin-scraper/src/main/java/org/pinkcrazyunicorn/quickie/plugins/scraper/HensslerScraper.java#L149}{\code{getRecipeURLs}} verwendet wird. Des Weiteren verwendet diese Methode keine der anderen Methoden der Klasse und auch keine andere Methode der Klasse verwendet diese. Diese Methode trägt zu niedriger Kohäsion in der Klasse bei. Dieser Code Smell sollte behoben werden, indem die Funktionalität in eine weitere Klasse ausgelagert wird.
```

Comment:
keine andere Methode der Klasse verwendet diese. Diese Methode trägt zu niedriger Kohäsion in der Klasse bei. -> keine andere Methode der Klasse verwendet diese, daher trägt diese Methode zu niedriger Kohäsion in der Klasse bei.